### PR TITLE
Fix wrong resourceName in codegen for metrics api objects

### LINKED ISSUE #93853: Wrong resourceName of metrics server api object in code-gen fail fake clientset
**What happened**:
`+resourceName` of `NodeMetricsList` and `PodMetrics` in `k8s.io/metrics` are set to `nodes` and `pods` respectively in codegen comments. For example, this makes generated fake client to use `nodes` as gvr to query, but the actually gvr should be `nodemetricses`, so `List()`/`Get()`/`Watch()` will always return not found. 

**What you expected to happen**:
`+resourceName` should be `nodemetricses` and `podmetricses`

**How to reproduce it (as minimally and precisely as possible)**:
```go
package main

import (
	"fmt"

	v1 "k8s.io/api/core/v1"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	metricsv1beta1 "k8s.io/metrics/pkg/apis/metrics/v1beta1"

	"k8s.io/apimachinery/pkg/api/resource"
	"k8s.io/metrics/pkg/client/clientset/versioned/fake"
)

func main() {
	fakeClient := fake.NewSimpleClientset(&metricsv1beta1.NodeMetrics{
		ObjectMeta: metav1.ObjectMeta{Name: "machine1"},
		Usage: v1.ResourceList{
			v1.ResourceCPU:    resource.MustParse("100m"),
			v1.ResourceMemory: resource.MustParse("1Gi"),
		},
	})
	fmt.Println(fakeClient.MetricsV1beta1().NodeMetricses().List(metav1.ListOptions{})) // &NodeMetricsList{ListMeta:{   <nil>},Items:[]NodeMetrics{},} <nil>
}

```

If you change `$GOPATH/pkg/mod/k8s.io/metrics/pkg/client/clientset/versioned/typed/metrics/v1beta1/fake/fake_nodemetrics.go`

```diff
- var nodemetricsesResource = schema.GroupVersionResource{Group: "metrics.k8s.io", Version: "v1beta1", Resource: "nodes"}
+ var nodemetricsesResource = schema.GroupVersionResource{Group: "metrics.k8s.io", Version: "v1beta1", Resource: "nodemetricses"}
```

you will get 
```go
fmt.Println(fakeClient.MetricsV1beta1().NodeMetricses().List(metav1.ListOptions{})) // &NodeMetricsList{ListMeta:{   <nil>},Items:[]NodeMetrics{NodeMetrics{ObjectMeta:{machine1      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []},Timestamp:0001-01-01 00:00:00 +0000 UTC,Window:{0s},Usage:k8s_io_api_core_v1.ResourceList{cpu: {{100 -3} {<nil>} 100m DecimalSI},memory: {{1073741824 0} {<nil>} 1Gi BinarySI},},},},} <nil>
```

**Anything else we need to know?**:

I've created an issue to resolve this issue

### PR DESCRIPTION
**Pull request by secret104278**
Signed-off-by: Yu Chih <secret104278@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
see https://github.com/kubernetes/kubernetes/issues/93853

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/93853

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by secret104278** [comment]
I signed it

File: staging/src/k8s.io/metrics/pkg/client/clientset/versioned/typed/metrics/v1alpha1/nodemetrics.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func newNodeMetricses(c *MetricsV1alpha1Client) *nodeMetricses {
 func (c *nodeMetricses) Get(ctx context.Context, name string, options v1.GetOptions) (result *v1alpha1.NodeMetrics, err error) {
 	result = &v1alpha1.NodeMetrics{}
 	err = c.client.Get().
-		Resource("nodes").
+		Resource("nodemetricses").
```
Comment (liggitt) [inline staging/src/k8s.io/metrics/pkg/client/clientset/versioned/typed/metrics/v1alpha1/nodemetrics.go:62]: I'm concerned about the client change here... this doesn't seem compatible

**Comment by liggitt** [comment]
This is an issue with the client-go fake client object tracker. Objects whose resourcenames differ from the naive pluralization guess must be added to an object tracker using a create call.

See comment at https://github.com/kubernetes/kubernetes/issues/93853#issuecomment-671615829

**Pull request closed without merging**